core/src/test/java/org/pentaho/platform/dataaccess/datasource/api/MetadataServiceTest.java

```diff
@@ -147,12 +150,11 @@ public void testGetMetadataDatasourceIds() throws Exception {
 
   @Test
   public void testGetMetadataDatasourceIdsError() throws Exception {
-    InterruptedException mockInterruptedException = mock( InterruptedException.class );
-    doThrow( mockInterruptedException ).when( metadataService ).sleep( 100 );
-
-    metadataService.getMetadataDatasourceIds();
-
+    doReturn( null ).when( metadataService.metadataDomainRepository ).getDomainIds();
+    doReturn( 1 ).when( metadataService ).noOfThreads();
```

Comment:
similar comment as before
I wouldn't mock out functions for the class under test, with the current implementation, it will get non-zero value regardless.